Exercise10.R

```diff
@@ -0,0 +1,65 @@
+##task 1
+
+#load packages
+library('ggplot2')
+
+#load the raw data
+data <- read.table("UWvMSU_1-22-13.txt",sep = "\t", header = TRUE)
+
+#change single scores to cumulative scores
+uw_cumulative_score <- 0
+msu_cumulative_score <- 0
+
+for(i in 1:nrow(data)){
+  if(data[i,"team"] == "UW"){
+    data[i,"score"] <- data[i,"score"] + uw_cumulative_score
+    uw_cumulative_score <- data[i,"score"]
+  }else if(data[i,"team"] == "MSU"){
+    data[i,"score"] <- data[i,"score"] + msu_cumulative_score
+    msu_cumulative_score <- data[i,"score"]
+  }
+}
+
+#add an extra row in case that a team doesn't score for a long time at the end
+uw_end <- data.frame(matrix(nrow = 1, ncol = 3))
+colnames(uw_end) <- colnames(uw_score)
+uw_end[1,"time"] <- 40
+uw_end[1,"score"] <- uw_cumulative_score
+
+msu_end <- data.frame(matrix(nrow = 1, ncol = 3))
+colnames(msu_end) <- colnames(msu_score)
+msu_end[1,"time"] <- 40
+msu_end[1,"score"] <- msu_cumulative_score
+
+uw_score <- rbind(data, uw_end)
+msu_score <- rbind(data, msu_end)

```

Comment:
an alternative way to do this:
# load table of scoring
scoring=read.table("UWvMSU_1-22-13.txt",header=TRUE,sep="\t",stringsAsFactors=FALSE)

# look at data
dim(scoring)
head(scoring)

# preallocating matrix to store cumulative scores
cum_scores=matrix(NA,nrow(scoring)+1,3)
cum_scores[,1]=c(0,scoring[,1])
cum_scores[1,2:3]=0
colnames(cum_scores)=c("time","UW","MSU")

# looping through individual scoring events
for(i in 1:nrow(scoring)){
  if(scoring[i,2]=="UW"){
    cum_scores[(i+1),2]=cum_scores[i,2]+scoring[i,3]
    cum_scores[(i+1),3]=cum_scores[i,3]
  }else{
    cum_scores[(i+1),2]=cum_scores[i,2]
    cum_scores[(i+1),3]=cum_scores[i,3]+scoring[i,3]
  }
}